dev_apps/uitest-privileged/js/API/settings.js

```diff
@@ -0,0 +1,47 @@
+var settingsTest = {
+  init: function () {
+    this.statusPlaceholder = document.getElementById('status-placeholder');
+    this.statusContainer = document.getElementById('status-container');
+
+    this.errorPlaceholder = document.getElementById('error-placeholder');
+    this.errorContainer = document.getElementById('error-container');
+  },
+  addStatus: function (aStatus) {
+    this.statusPlaceholder.innerHTML = this.statusPlaceholder.innerHTML + "<br\>" + aStatus;
+  },
+  onError: function(aError) {
+    console.error('ERROR ' + aError.name || aError);
+    this.errorContainer.classList.remove('hidden');
+    this.errorPlaceholder.textContent = aError.name;
+  },
+  runTest: function() {
+    if (navigator.mozSettings)
```

Comment:
nit: maybe we should bail early and not do the rest of the logic in the body. Like:
if (!navigator.mozSettings) {
  this.onError(...);
  return;
}

Just to make it easier to read.